providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/resolver/process/storage/connector/file/FileConnector.java

```diff
@@ -9,36 +9,73 @@
 import java.io.IOException;
 import java.nio.charset.StandardCharsets;
 import java.nio.file.Files;
+import java.nio.file.Path;
 import java.nio.file.Paths;
 import java.util.concurrent.BlockingQueue;
 import java.util.concurrent.LinkedBlockingQueue;
 
 /**
- * File connector reads flag configurations and expose the context through {@code Connector} contract.
+ * File connector reads flag configurations from a given file, polls for changes and expose the content through
+ * {@code Connector} contract.
  * The implementation is kept minimal and suites testing, local development needs.
  */
 @SuppressFBWarnings(value = {"EI_EXPOSE_REP", "PATH_TRAVERSAL_IN"},
         justification = "File connector read feature flag from a file source.")
 @Slf4j
 public class FileConnector implements Connector {
 
+    private static final int POLL_INTERVAL_MS = 5000;
+
     private final String flagSourcePath;
     private final BlockingQueue<StreamPayload> queue = new LinkedBlockingQueue<>(1);
+    private boolean shutdown = false;
 
     public FileConnector(final String flagSourcePath) {
         this.flagSourcePath = flagSourcePath;
     }
 
     /**
-     * Initialize file connector. Reads content of the provided source file and offer it through queue.
+     * Initialize file connector. Reads file content, poll for changes and offer content through the queue.
      */
     public void init() throws IOException {
-        final String flagData = new String(Files.readAllBytes(Paths.get(flagSourcePath)), StandardCharsets.UTF_8);
+        Thread watcherT = new Thread(() -> {
```

Comment:
I could be missing it, but taking a quick look, it seems like we don't have the necessary synchronization here, or in [FlagStore](https://github.com/open-feature/java-sdk-contrib/blob/main/providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/resolver/process/storage/FlagStore.java) to prevent 2 threads from calling this at nearly the same time and starting multiple instances of this thread.

Maybe we can implement something like that in the provider, or in the `FlagStore` so it doesn't need to be implemented in every `Connector`?

Reply:
Good point. I didn't consider a race condition when implementing this. The reason was not expecting the initializer to be called concurrently. But a quick walkthrough on potential usage patterns, there could be race conditions if the same provider is registered with multiple names. I think I would like to revisit this topic in a dedicated PR.

Reply:
I created follow-up tickets for Go (https://github.com/open-feature/go-sdk-contrib/issues/417) & Java (https://github.com/open-feature/java-sdk-contrib/issues/623)